src/main/java/com/gluonhq/RunAgentMojo.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021, Gluon
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.gluonhq;
+
+import com.gluonhq.attach.AttachService;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Profile;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.shared.invoker.*;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+@Mojo(name = "run", requiresDependencyResolution = ResolutionScope.RUNTIME)
+@Execute(phase = LifecyclePhase.PROCESS_CLASSES)
+public class RunAgentMojo extends NativeBaseMojo {
+
+    private static final String POM_XML = "pom.xml";
```

Comment:
why don't you use:
```
@Parameter(readonly = true, required = true, defaultValue = "${basedir}/pom.xml")
    String pom;

@Parameter(readonly = true, required = true, defaultValue = "${basedir}/runPom.xml")
    String runPom;
```
?

Reply:
Adding `@Parameter` to an instance variable enables Maven to set the value via `pom.xml`. Since, we do not want these variables to be configured via pom.xml, there is no advantage of adding `@Parameter` to them.

Reply:
`readonly` prevents any further configuration

Reply:
`readonly` is used when we want to avoid a user to set a parameter's value directly. However, it should be dependent on some other value which could be changed.

The question is what do we gain by annotating `@Parameter` to the variable?

Reply:
Not much, I agree, but I guess we follow the Maven pattern where at some point things can be configured externally (i.e a pom not named `pom.xml` or located in a different folder).

Reply:
Maven introduced `@Parameter` to make it easier to introduce configurable properties. If we do not want to configure the values of `pom` or `runPom` via xml or command line, I don't see any value in adding the `@Parameter` annotation to these variables.

Reply:
Ok